src/display/api.js

```diff
@@ -957,10 +957,41 @@ var PDFPageProxy = (function PDFPageProxyClosure() {
      */
     getTextContent: function PDFPageProxy_getTextContent(params) {
       params = params || {};
-      return this.transport.messageHandler.sendWithPromise('GetTextContent', {
+      var readableStream =
+        this.transport.messageHandler.sendWithStream('GetTextContent', {
         pageIndex: this.pageNumber - 1,
         normalizeWhitespace: (params.normalizeWhitespace === true),
         combineTextItems: (params.disableCombineTextItems !== true),
+      }, {
+        highWaterMark: 5,
+        size(textContent) {
+          return textContent.items.length;
+        }
+      });
+      return new Promise((resolve, reject) => {
+        function pump() {
+          reader.read().then((result) => {
+            if (result.done) {
+              resolve(textContent);
+            } else {
+              for (var key in result.value.styles) {
+                textContent.styles[key] = result.value.styles[key];
+              }
+              textContent.items = textContent.items.concat(result.value.items);
```

Comment:
Also, as mentioned in https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/concat, `concat` creates a *new* array.
Hence, a better approach is probably to use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/push, which just extends the existing array instead.

Hence it's probably not a bad idea to use `Util.appendToArray` instead here!